src/prefect/serialization/state.py

```diff
@@ -36,10 +36,10 @@ class Meta:
 
     context = fields.Dict(key=fields.Str(), values=JSONCompatible(), allow_none=True)
     message = fields.String(allow_none=True)
-    _result = Nested(StateResultSchema, allow_none=False, value_selection_fn=get_safe)
+    _result = Nested(StateResultSchema, allow_none=False, value_selection_fn=None)
```

Comment:
On a pairing call with @cicdw there was an implication that somewhere around here (or maybe it was just that his memory about it was triggered around here) we need to be worried about maintaining deserialization of old style `State._result` in pipeline logic, so calling that out here in case we want to have the discussion at this location.